sentinel-2/ndvi_difference/script.js

```diff
@@ -0,0 +1,94 @@
+//VERSION=3
+// Script to extract NDVI difference between the latest acquisition and the acquisition 10-day prior to the latest within a specified time range
+function setup() {
+    return {
+      input: [{
+        bands: ["B04", "B08", "SCL", "dataMask"],
+        units: "DN"
+      }],
+      output: [
+        {id: "default", bands: 4},
+        {id: "index", bands: 1, sampleType: "FLOAT32"},
+        {id: "eobrowserStats", bands: 2, sampleType: "FLOAT32"},
+        {id: "dataMask", bands: 1}
+      ],
+      mosaicking: Mosaicking.ORBIT
+    }
+    
+  }
+  
+  function evaluatePixel(samples) {
+    // ndvi difference
+    let latest = samples[0];
+    let prior = samples[1];
+    let combineMask = latest.dataMask * prior.dataMask;
+    const diff = combineMask === 1 ? index(latest.B08, latest.B04) - index(prior.B08, prior.B04) : NaN;
+    
+    // visualisation
+    const ramps = [
+        [-2, 0x000000],
+        [2, 0xffffff]
+    ]
+    const visualizer = new ColorRampVisualizer(ramps);
+    let imgVals = visualizer.process(diff);
```

Comment:
`diff` can be NaN if the `combineMask` is 0. I tested it and `visualizer.process(NaN)` returns `[1,1,1]`. I think it would be good to explicitly catch that and visualize a No Data value

Reply:
True, that should be enough

Reply:
Shall I merge it?